runtime/lua/vim/lsp/handlers.lua

```diff
@@ -149,6 +149,31 @@ M['workspace/applyEdit'] = function(_, _, workspace_edit)
   }
 end
 
+--@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_configuration
+M['workspace/configuration'] = function(err, _, params, client_id)
+  local client = vim.lsp.get_client_by_id(client_id)
+  if not client then
+    err_message("LSP[id=", client_id, "] client has shut down after sending the message")
+  end
+  if err then error(vim.inspect(err)) end
+  if not params.items then
+    return {}
+  end
+
+  local result = {}
+  for _, item in ipairs(params.items) do
+    if item.section then
+      local value = util.lookup_section(client.config.settings, item.section) or vim.NIL
+      -- For empty sections with no explicit '' key, return settings as is
+      if value == vim.NIL and item.section == '' then
+        value = client.config.settings or vim.NIL
```

Comment:
shouldn't it return straightaway in that case ? since it covers all of the config.  It doesn't hurt to go on so it may be ok.

Reply:
I guess there is the corner case to worry about, where in the array of ConfigurationParams, the first param requests an item.section that is empty (adding client.config.settings to the table), and the second ConfigurationParams requests one that is not empty (adding a specific setting to the table).

something like:
```lua
params = {
    items = { 
        { section = ''}
        { section = 'python.analysis.autoSearchPaths'}
    }
}
```

I don't know if there *is* a server that does this (it would probably be a bug), but I think we should respond to the entire list of items.